### ISSUE TITLE
[QUESTION] Headful Chrome on Docker for Linux - Why is the screencast working on Docker for Mac, but not Docker for Linux?

### ISSUE THREAD

**Issue by rwoll** [microsoft/playwright#8422]
I have a test in a container that work on Docker for Mac, but if I run the same setup in Docker for Linux, the code fails. Can you help spot the difference? (I'm assuming it's a missing dependency somewhere.)

Try running the following code on macOS with Docker installed, and then doing the same on a Linux host with Docker installed:
```
$ git clone https://github.com/rwoll/playwright-screencast-next-repro.git
$ cd playwright-screencast-next-repro
$ ./run_tests.sh
```

On mac, both tests will pass; but on Linux, they fail like:

```
=======================================================================
BUILDING 14 (logs are being saved to build_14.log)
=======================================================================
=======================================================================
TESTING 14
=======================================================================
Using config at /opt/test/playwright.config.ts

Running 1 test using 1 worker

  ✘  screencast.spec.ts:3:1 › should produce screencast frame (30s)


  1) screencast.spec.ts:3:1 › should produce screencast frame ======================================

    Timeout of 30000ms exceeded.

    Pending operations:
      - cDPSession.send at screencast.spec.ts:8:13

    cDPSession.send: Protocol error (Page.startScreencast): Target closed.

       6 |     cdp.once("Page.screencastFrame", res);
       7 |   });
    >  8 |   await cdp.send("Page.startScreencast");
         |             ^
       9 |   await page.goto('https://playwright.dev/');
      10 |   await screencastFrame;
      11 | });

        at /opt/test/screencast.spec.ts:8:13
        at WorkerRunner._runTestWithBeforeHooks (/opt/test/node_modules/@playwright/test/lib/test/workerRunner.js:426:7)

  Slow test: screencast.spec.ts (30s)

  1 failed
    screencast.spec.ts:3:1 › should produce screencast frame =======================================
```

Additional notes:
* Linux Host is Amazon Linux 2 AMI (linux/amd64) running Docker Version 20.10.4
* Mac Host is macOS 11.4 (darwin/amd64) running Docker CE 20.10.7
* headless mode works fine in both places as does PW chromium.

Thanks!

**Comment by pavelfeldman** [comment]
I don't think we support headed scenarios in Docker, so you are on your own in this scenario...

**Comment by rwoll** [comment]
Noted! Thanks anyways!

**Comment by rwoll** [comment]
I got this all working by using an alternate GL implementation.

The example repo has been updated. 

Now, when I run it, the projects with the new arg pass and the projects without it fail on GPU-less Docker for Linux containers as expected. 🎉 

NB: For anyone who may stumble on this in the future: that example repo is by no means optimized nor production code.

@pavelfeldman Anecdotally, how does swiftshader compare to the default? Any caveats you're aware of?
